solr/core/src/java/org/apache/solr/response/QueryResponseWriter.java

```diff
@@ -39,23 +41,26 @@
  * <p>A single instance of any registered QueryResponseWriter is created via the default constructor
  * and is reused for all relevant queries.
  */
+@ThreadSafe
 public interface QueryResponseWriter extends NamedListInitializedPlugin {
   public static String CONTENT_TYPE_XML_UTF8 = "application/xml; charset=UTF-8";
   public static String CONTENT_TYPE_TEXT_UTF8 = "text/plain; charset=UTF-8";
   public static String CONTENT_TYPE_TEXT_ASCII = "text/plain; charset=US-ASCII";
 
   /**
-   * Write a SolrQueryResponse, this method must be thread save.
-   *
-   * <p>Information about the request (in particular: formatting options) may be obtained from
-   * <code>req</code> but the dominant source of information should be <code>rsp</code>.
-   *
-   * <p>There are no mandatory actions that write must perform. An empty write implementation would
-   * fulfill all interface obligations.
+   * Writes the response to the {@link OutputStream}. {@code contentType} is from {@link
+   * #getContentType(SolrQueryRequest, SolrQueryResponse)}, and it's often ignored.
    */
-  public void write(Writer writer, SolrQueryRequest request, SolrQueryResponse response)
+  void write(
+      OutputStream out, SolrQueryRequest request, SolrQueryResponse response, String contentType)
```

Comment:
Added contentType since the production code path will have this already, and we'll need it in TextQueryResponseWriter

Reply:
It's a debatable change... it'd be less disruptive to leave it as it was and then only the text formats will re-acquire it which is fairly trivial parsing to do a second time.  Shrug.